docs/HowToFAQ/FAQMultClauses.md

```diff
@@ -0,0 +1,18 @@
+---
+title: When a lemma has multiple ensures clauses, I’m finding that they interact, when I expected them to be independent.  For example, commenting out one of them can make another one not verify.  How should I think about this?
+---
+
+## Question
+
+When a lemma has multiple ensures clauses, I’m finding that they interact, when I expected them to be independent.  For example, commenting out one of them can make another one not verify.  How should I think about this?
+
+## Answer
+
+Multiple ensures clauses, such as `ensures P ensures Q ensures R` is equivalent to the conjunction, in order, of the ensures predicates: `ensures P && Q && R`.
+The order is important if an earlier predicate is needed to be sure that a later one is well defined. For example the two predicates in `ensures a != null ensures a.Length > 0`
+must be in that order because `a.Length` is well-defined only if `a != null`.
```

Comment:
Arrays cannot be null; may we can give an example with an index and an array?

Reply:
Sure they can, that is variables of array?<> type can be null:
```
method ma(a: array?<int>) {
  if a.Length == 0 { } // ERROR - a might be null
}
```

Reply:
Sure, but that's `array?`, not `array`.  But because in previous versions `array` (without `?`) could be null as well, we find leftover docs from time to time that still check for `null` arrays; so I was suggesting to use a value type instead to not add to the confusion.

Reply:
I clarified that `a` is an `array?`